homeassistant/components/octoprint/__init__.py

```diff
@@ -51,10 +54,32 @@ def ensure_valid_path(value):
     return value
 
 
+class OctoprintBinarySensorMetadata(NamedTuple):
+    """Metadata for an individual Skybell binary_sensor."""
+
+    name: str
+    endpoint: str
+    group: str
+    key: str
+    unit: str | None
+
+
 BINARY_SENSOR_TYPES = {
     # API Endpoint, Group, Key, unit
-    "Printing": ["printer", "state", "printing", None],
-    "Printing Error": ["printer", "state", "error", None],
+    "Printing": OctoprintBinarySensorMetadata(
+        "entry",
+        endpoint="printer",
+        group="state",
+        key="printing",
+        unit=None,
```

Comment:
If you add `None` as default parameter for `unit_of_measurement`, you can remove it here (and the one below).

Reply:
Done, but I did because I am looking for a pattern.